### PR TITLE
Harmonise component versions and remove redundant conditionals.

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: nix/supervisord-cluster/topology/cardano-topology.cabal [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cabal-version: 2.4
 
 name:                  cardano-topology
-version:               1.25.0
+version:               1.6.0
```
Comment (Jimbo4350) [inline nix/supervisord-cluster/topology/cardano-topology.cabal:4]: Should this be 1.26.0?

File: cardano-api/test/cardano-api-test.cabal [hunk 2]
```diff
@@ -18,17 +18,12 @@ common project-config
...
                         -Wincomplete-record-updates
                         -Wincomplete-uni-patterns
-                        -Wredundant-constraints
                         -Wpartial-fields
-                        -Wcompat
-
-  if impl(ghc >= 8.10)
-    ghc-options:        -Wunused-packages
-
-  if impl(ghc < 8.10)
```
Comment (Jimbo4350) [inline cardano-api/test/cardano-api-test.cabal:30]: Should we update the docs that we expect >= 8.10?

**Review by Jimbo4350 — COMMENTED** [review]
Couple of comments to address

File: cardano-api/test/cardano-api-test.cabal (hunk 2 above)
Comment (newhoggy) [reply cardano-api/test/cardano-api-test.cabal:30]: Actually, yeah.

File: nix/supervisord-cluster/topology/cardano-topology.cabal (hunk 1 above)
Comment (newhoggy) [reply nix/supervisord-cluster/topology/cardano-topology.cabal]: Oops.  Fixed!

**Comment by newhoggy** [comment]
bors merge

File: cardano-api/test/cardano-api-test.cabal (hunk 2 above)
Comment (newhoggy) [reply cardano-api/test/cardano-api-test.cabal:30]: I've updated one place I've found in the documentation.

File: doc/getting-started/install.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ To download the source code and build it, you need the following packages and to
 * developer libraries for `ncurses`,
 * `ncurses` compatibility libraries,
 * the Haskell build tool `cabal`,
-* the GHC Haskell compiler.
+* the GHC Haskell compiler (version `8.10.2` or above).
```
Comment (newhoggy) [inline doc/getting-started/install.md:30]: This is the place.

**Review by erikd — APPROVED** [review]

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**